src/enc_effects.c

```diff
@@ -0,0 +1,366 @@
+
+    if (ENCOUNTER_EFFECTS[effectsID].cutinVFX == CATCHALL_ENC) {
+        return ov5_021DEEC8(battleData);
+    } else {
+        return ENCOUNTER_EFFECTS[effectsID].cutinVFX;
+    }
+}
+
+static u32 EncEffects_GetBGMFromEffects (u32 effectsID, const UnkStruct_ov6_02240D5C * battleID)
+{
+    GF_ASSERT(effectsID < ENCFX_NUM_ENTRIES);
+    return ENCOUNTER_EFFECTS[effectsID].bgmSEQID;
+}
+
+u32 EncEffects_GetCutInEffectForBattle (const UnkStruct_ov6_02240D5C * battleData)
+{
+    u32 encEffects = EncEffects_GetEffectsForBattle(battleData);
+    return EncEffects_GetCutInFromEffects(encEffects, battleData);
+}
+
+u32 EncEffects_GetBGMForBattle (const UnkStruct_ov6_02240D5C * battleData)
+{
+    u32 encEffects = EncEffects_GetEffectsForBattle(battleData);
+    return EncEffects_GetBGMFromEffects(encEffects, battleData);
+}
+
+static u32 EncEffects_GetEffectsForTrainerClass (u32 trainerClass)
+{
+    switch (trainerClass) {
+        // Gym leaders
+        case 62:    return ENCFX_OREBURGH_GYM_LEADER;   // Roark
+        case 74:    return ENCFX_ETERNA_GYM_LEADER;     // Gardenia      
+        case 75:    return ENCFX_PASTORIA_GYM_LEADER;   // Crasher Wake
+        case 76:    return ENCFX_VEILSTONE_GYM_LEADER;  // Maylene
+        case 77:    return ENCFX_HEARTHOME_GYM_LEADER;  // Fantina
+        case 78:    return ENCFX_SNOWPOINT_GYM_LEADER;  // Candice
+        case 64:    return ENCFX_CANALAVE_GYM_LEADER;   // Byron
+        case 79:    return ENCFX_SUNYSHORE_GYM_LEADER;  // Volkner
+
+        // Elite Four
+        case 65:    return ENCFX_ELITE_FOUR_1;          // Aaron
+        case 66:    return ENCFX_ELITE_FOUR_2;          // Bertha
+        case 67:    return ENCFX_ELITE_FOUR_3;          // Flint
+        case 68:    return ENCFX_ELITE_FOUR_4;          // Lucian
+
+        // Champion
+        case 69:    return ENCFX_CHAMPION;              // Cynthia
+
+        // Rival
+        case 63:    return ENCFX_RIVAL;                 // Barry/Cedric
+
+        // Team Galactic
+        case 73:    return ENCFX_GALACTIC_GRUNT;        // Grunt (Male)
+        case 89:    return ENCFX_GALACTIC_GRUNT;        // Grunt (Female)
+        case 72:    return ENCFX_GALACTIC_COMMANDER;    // Mars
+        case 87:    return ENCFX_GALACTIC_COMMANDER;    // Jupiter
+        case 88:    return ENCFX_GALACTIC_COMMANDER;    // Saturn
+        case 86:    return ENCFX_GALACTIC_LEADER;       // Cyrus
+
+        // Frontier Brains
+        case 97:    return ENCFX_FRONTIER_BRAIN;        // Palmer
+        case 99:    return ENCFX_FRONTIER_BRAIN;        // Argenta
+        case 100:   return ENCFX_FRONTIER_BRAIN;        // Thorton
+        case 101:   return ENCFX_FRONTIER_BRAIN;        // Dahlia
+        case 102:   return ENCFX_FRONTIER_BRAIN;        // Darach
+
+        // All other trainers
+        default:    return ENCFX_GENERIC_TRAINER;
+    }
+}
+
+// the party passed in here is actually the *enemy* party
+// that tripped me up for a little bit
```

Comment:
Personal anecdotes are fun but don't really belong in the code. Feel free to rant in the commit or PR description.
Maybe rename `wildParty` to `enemyParty`?